src/components/configure/AddExtraConfigDialog.tsx

```diff
@@ -0,0 +1,125 @@
+import { TextField } from '../TextField';
+
+export interface DialogResult {
+  config: string;
+  value: string;
+}
+
+export default class AddExtraConfigDialog extends ReactWidget implements Dialog.IBodyWidget<DialogResult | null> {
+  clusterName: string;
+  selectedConfigBundles: string[];
+  selectedConfig?: string;
+  configValue?: string;
+
+  constructor(clusterName: string, selectedConfigBundles: string[]) {
+    super();
+    this.clusterName = clusterName;
+    this.selectedConfigBundles = selectedConfigBundles;
+  }
+
+  render() {
+    return <_AddExtraConfigDialog clusterName={this.clusterName} selectedConfigBundles={this.selectedConfigBundles} onSetConfig={s => (this.selectedConfig = s)} onSetValue={s => (this.configValue = s)} />;
+  }
+
+  getValue() {
+    if (!this.selectedConfig || !this.configValue) {
+      return null;
+    }
+
+    return {
+      config: this.selectedConfig,
+      value: this.configValue
+    };
+  }
+}
+
+interface MyProps {
+  clusterName: string;
+  selectedConfigBundles: string[];
+  onSetConfig: (config: string) => void;
+  onSetValue: (value: string) => void;
+}
+
+function _AddExtraConfigDialog({ clusterName, selectedConfigBundles, onSetConfig, onSetValue }: MyProps) {
+  const configOptions = UIStore.useState(s => s.configOptions);
+  const groupedConfigOptions = useMemo(() => {
+    const map = configOptions.reduce((acc, curr) => {
+      const cat = curr.category ?? 'Other';
+      if (!(cat in acc)) {
+        acc[cat] = [];
+      }
+      acc[cat].push(curr);
+      return acc;
+    }, {} as any);
+
+    return Object.keys(map).map(category => ({
+      category,
+      options: map[category].map((c: any) => ({ label: c.name, value: c.name }))
+    }));
+  }, [configOptions]);
+
+  const configBundles = UIStore.useState(s => s.configBundleOptions);
+  const configuredOptionsFromBundle = useMemo(() => {
+    const map: any = {};
+    configBundles
+      .filter(b => selectedConfigBundles.includes(b.name))
+      .filter(b => !b.clusterFilter || b.clusterFilter.includes(clusterName))
+      .forEach(bundle => {
+        bundle.options.forEach(option => (map[option.name] = bundle.displayName));
+      });
+    return map;
+  }, [configBundles, clusterName, selectedConfigBundles]);
+
+  const [configOption, setConfigOption] = useState<{ label: String; value: string }>();
+  useEffect(() => {
+    if (configOption) onSetConfig(configOption.value);
+  }, [configOption]);
+
+  const [value, setValue] = useState<string>('');
+  useEffect(() => {
+    onSetValue(value);
+  }, [value]);
+
+  const overridenConfigBundle = useMemo(() => {
+    if (!configuredOptionsFromBundle || !configOption) return;
+    return configuredOptionsFromBundle[configOption.value];
+  }, [selectedConfigBundles, configuredOptionsFromBundle, configOption]);
+
+  const formatGroupLabel = (data: any) => (
+    <div>
+      <span>{data.category}</span>
+    </div>
+  );
+
+  return (
+    <div style={{ width: 400 }}>
```

Comment:
The unit is `px`, and `400px` just an arbitrary number that looks nice in my opinion.

Reply:
I haven't tried running the extension yet. What happens if you change the with of the panels in the UI? Or you managed to prevent that? I'm interested in knowing what happens when they are both smaller and larger.
(just wondering if we need to have a min-width instead of just width for example).

Reply:
JupyterLab itself breaks down below 600px, so I don't think 400px will be a problem 

<img width="714" alt="Screenshot 2023-07-30 at 17 31 38" src="https://github.com/swan-cern/sparkconnect/assets/4845621/24d75729-ea1b-4259-bf69-abd0f4e7f148">

Reply:
But the panel can be resized to be less than that?
Maybe it would be quicker to just try it... Can you put all the latest stuff in a dev machine and let us know?

Reply:
I can, but my dev machine only allows my credentials I suppose, since the Kerberos token to access EOS uses my personal token. At least that's what I understand